### DISCUSSION TITLE
Why is my form validation not triggered when the form is out of sight but I am still in the form element?

### DISCUSSION THREAD

**Discussion by manuelpoelzl** [react-hook-form/react-hook-form#9399] [Q&A] [answered]
Hello Community!

I have a question regarding the form validation.
Specifically I have the following scenario, I have a TabPanel in my form where I can switch between "General" and "Advanced", as I said this panel is inside the form.

Now the situation is that, for example, the fields in my General tab are required, and the fields in my Advanced tab.
If I stay on the first initial tab and try to send the form without the required values, the errors are displayed as intended. However, if I switch to the second tab and send the form, also without the required values, then the validation is not performed and I can send the form with missing values.

For better understanding, I have prepared a [CodeSandbox sample](https://codesandbox.io/s/busy-flower-3t1vpo?file=/src/App.tsx) that shows the problem.

Every field in my form in the sample is required.
The following steps are sufficient for reproducing:

1. stay in the General tab and click the Submit button - the errors are displayed.
2. leave the field in the General tab empty and switch to the second tab and enter some text in the field and then click the Submit button - no validation is performed.

I would be very grateful if you could give me a solution here!

**Comment by leapful** [comment, marked as answer]
@manuelpoelzl 

Prime React TabView component only renders active TabPanel by default while your implementation is using built-in validation of react-hook-form which sets validators based on each rendered field element therefore when one TabPanel is hidden, Prime React unmount the whole content including validators of react-hook-form inside it. 

Just switch to mode render all tab panels to solve the issue.

Working demo: https://codesandbox.io/s/nostalgic-danilo-ntcvn1

**Reply by manuelpoelzl** [reply]
You are a genius! Thank you very much! 🥳
